hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/JDBCSource.java

```diff
@@ -0,0 +1,233 @@
+          .getString(0);
+      LOG.info("Sending {} with checkpoint val {} ", incrementalColumn, max);
+      return Pair.of(Option.of(rowDataset), max);
+    } else {
+      return Pair.of(Option.of(rowDataset), null);
+    }
+  }
+
+
+  protected static class Config {
+
+    /**
+     * {@value #URL} is the jdbc url for the Hoodie datasource
+     */
+    private static final String URL = "hoodie.datasource.jdbc.url";
+
+    private static final String URL_PROP = "url";
+
+    /**
+     * {@value #USER} is the username used for JDBC connection
+     */
+    private static final String USER = "hoodie.datasource.jdbc.user";
+
+    /**
+     * {@value #USER_PROP} used internally to build jdbc params
+     */
+    private static final String USER_PROP = "user";
+
+    /**
+     * {@value #PASSWORD} is the password used for JDBC connection
+     */
+    private static final String PASSWORD = "hoodie.datasource.jdbc.password";
+
+    /**
+     * {@value #PASSWORD_FILE} is the base-path for the JDBC password file
+     */
+    private static final String PASSWORD_FILE = "hoodie.datasource.jdbc.password.file";
+
+    /**
+     * {@value #PASSWORD_PROP} used internally to build jdbc params
+     */
+    private static final String PASSWORD_PROP = "password";
+
+    /**
+     * {@value #DRIVER_CLASS} used for JDBC connection
+     */
+    private static final String DRIVER_CLASS = "hoodie.datasource.jdbc.driver.class";
+
+    /**
+     * {@value #DRIVER_PROP} used internally to build jdbc params
+     */
+    private static final String DRIVER_PROP = "driver";
+
+    /**
+     * {@value #RDBMS_TABLE_NAME} RDBMS table to pull
+     */
+    private static final String RDBMS_TABLE_NAME = "hoodie.datasource.jdbc.table.name";
+
+    /**
+     * {@value #RDBMS_TABLE_PROP} used internall for jdbc
+     */
+    private static final String RDBMS_TABLE_PROP = "dbtable";
+
+    /**
+     * {@value #INCREMENTAL_COLUMN} if ran in incremental mode, this field will be used to pull new data incrementally
+     */
+    private static final String INCREMENTAL_COLUMN = "hoodie.datasource.jdbc.table.incremental.column.name";
+
+    /**
+     * {@value #IS_INCREMENTAL} will the JDBC source do an incremental pull?
+     */
+    private static final String IS_INCREMENTAL = "hoodie.datasource.jdbc.incremental.pull";
+
+    /**
+     * {@value #INTERVAL} regular interval for which DeltaStreamer will be scheduled
+     */
+    private static final String INTERVAL = "hoodie.datasource.jdbc.table.incremental.pull.interval";
+
+    /**
+     * {@value #TIME_FORMAT} regular interval for which DeltaStreamer will be scheduled
+     */
+    private static final String TIME_FORMAT = "hoodie.datasource.jdbc.table.incremental.pull.interval";
+
+    /**
+     * {@value #TIME_FORMAT_PROP} used internally to build TIME_FORMAT for jdbc
+     */
+    private static final String TIME_FORMAT_PROP = "timestampFormat";
```

Comment:
some of these don't have defaults ?

Reply:
This had to be remove I forgot that. A better way to handle all this is through the extra.options in jdbc.properties file
`hoodie.datasource.jdbc.extra.options="fetchsize=1000,timestampFormat=yyyy-mm-dd hh:mm:ss"`